watchtower/wtdb/client_db.go

```diff
@@ -1525,14 +1545,37 @@ func (c *ClientDB) DeleteSession(id SessionID) error {
 			return err
 		}
 
-		// Get the acked updates range index for the session. This is
-		// used to get the list of channels that the session has updates
-		// for.
 		ackRanges := sessionBkt.NestedReadBucket(cSessionAckRangeIndex)
+
+		// There is a small chance that the session only contains rogue
+		// updates. In that case, there will be no ack-ranges index but
```

Comment:
newbie Q: why rogue updates must be equal to the `MaxUpdates`? Could there be a case, that we only have rogue updates, but the number doesn't reach the max?

Reply:
> Could there be a case, that we only have rogue updates, but the number doesn't reach the max?

Yes totally but then we want the rest of the function to continue as normal so that all those acked updates can be handled. Ie, only in the case where _all_ the updates are rogue updaters, can we just delete the session bucket right here and return.